scapy/contrib/automotive/scanner/test_case.py

```diff
@@ -0,0 +1,237 @@
+# This file is part of Scapy
+# See http://www.secdev.org/projects/scapy for more information
+# Copyright (C) Nils Weiss <nils@we155.de>
+# This program is published under a GPLv2 license
+
+# scapy.contrib.description = TestCase base class definitions
+# scapy.contrib.status = library
+
+
+from collections import defaultdict
+
+from scapy.compat import Any, Union, List, Optional, \
+    Dict, Tuple, Set, Callable, TYPE_CHECKING
+from scapy.utils import make_lined_table, SingleConversationSocket
+import scapy.modules.six as six
+from scapy.supersocket import SuperSocket
+from scapy.contrib.automotive.scanner.graph import _Edge
+from scapy.contrib.automotive.ecu import EcuState, EcuResponse
+
+
+if TYPE_CHECKING:
+    from scapy.contrib.automotive.scanner.configuration import AutomotiveTestCaseExecutorConfiguration  # noqa: E501
+
+
+if six.PY34:
+    from abc import ABC, abstractmethod
+else:
+    from abc import ABCMeta, abstractmethod
+    ABC = ABCMeta('ABC', (), {})  # type: ignore
```

Comment:
You can just do
`@six.add_metaclass(abc.ABCMeta)`
and you don't need all of that.

Reply:
Thanks a lot for this hint